ledger-core/virtual/integration/method_test.go

```diff
@@ -1704,3 +1726,194 @@ func TestVirtual_CallContractTwoTimes(t *testing.T) {
+			State:        contract.CallDirty,
+		}
+		fooIsolation = contract.MethodIsolation{
+			Interference: contract.CallTolerable,
+			State:        contract.CallDirty,
+		}
+
+		classA        = gen.UniqueGlobalRef()
+		objectAGlobal = reference.NewSelf(server.RandomLocalWithPulse())
+
+		objectBGlobal = reference.NewSelf(server.RandomLocalWithPulse())
+
+		callOutgoing = server.RandomLocalWithPulse()
+
+		outgoingCallRef = reference.NewRecordOf(objectBGlobal, callOutgoing)
+	)
+
+	Method_PrepareObject(ctx, server, payload.Ready, objectAGlobal)
+	Method_PrepareObject(ctx, server, payload.Ready, objectBGlobal)
+
+	// add ExecutionMocks to runnerMock
+	{
+		outgoingCall := execution.NewRPCBuilder(outgoingCallRef, objectBGlobal).
+			CallMethod(objectAGlobal, classA, "Bar", []byte("123")).
+			SetInterference(barIsolation.Interference).
+			SetIsolation(barIsolation.State)
+		objectBExecutionMock := runnerMock.AddExecutionMock("Foo")
+		objectBExecutionMock.AddStart(
+			func(ctx execution.Context) {
+				logger.Debug("ExecutionStart [B.Foo]")
+				assert.Equal(t, objectBGlobal, ctx.Object)
+				assert.Equal(t, callOutgoing, ctx.Request.CallOutgoing)
+			},
+			&execution.Update{
+				Type:     execution.OutgoingCall,
+				Error:    nil,
+				Outgoing: outgoingCall,
+			},
+		)
+		objectBExecutionMock.AddContinue(
+			func(result []byte) {
+				logger.Debug("ExecutionContinue [B.Foo]")
+				require.Equal(t, []byte("finish A.Bar"), result)
+			},
+			&execution.Update{
+				Type:   execution.Done,
+				Result: requestresult.New([]byte("finish B.Foo"), objectBGlobal),
+			},
+		)
+
+		runnerMock.AddExecutionMock("Bar").AddStart(
+			func(ctx execution.Context) {
+				logger.Debug("ExecutionStart [A.Bar]")
+				assert.Equal(t, objectAGlobal, ctx.Request.Callee)
+				assert.Equal(t, objectBGlobal, ctx.Request.Caller)
+				assert.Equal(t, []byte("123"), ctx.Request.Arguments)
+			},
+			&execution.Update{
+				Type:   execution.Done,
+				Result: requestresult.New([]byte("finish A.Bar"), objectBGlobal),
+			},
+		)
+
+		runnerMock.AddExecutionClassify("Bar", barIsolation, nil)
+		runnerMock.AddExecutionClassify("Foo", fooIsolation, nil)
+	}
+
+	callCount := 0
+
+	firstPulse := server.GetPulse().PulseNumber
+
+	{
+		typedChecker.VStateReport.SetResend(true)
+		typedChecker.VDelegatedCallRequest.SetResend(true)
+		typedChecker.VDelegatedCallResponse.SetResend(true)
+		typedChecker.VDelegatedRequestFinished.SetResend(true)
+		typedChecker.VCallRequest.Set(func(request *payload.VCallRequest) bool {
+			assert.Equal(t, objectAGlobal, request.Callee)
+			assert.Equal(t, objectBGlobal, request.Caller)
+			assert.Equal(t, []byte("123"), request.Arguments)
+			assert.Equal(t, payload.CTMethod, request.CallType)
+			assert.Equal(t, uint32(1), request.CallSequence)
+			assert.Equal(t, outgoingCallRef, request.CallReason)
+			assert.Equal(t, payload.BuildCallFlags(barIsolation.Interference, barIsolation.State), request.CallFlags)
```

Comment:
we test code around obtaining delegation token and attaching it to external calls. it's very strange that so many fields of VCallRequest are checked, but **ZERO** fields of VDelegatedCallRequest, VDelegatedCallResponse are checked. Also, VDelegatedRequestFinished is not checked for presence of the token.

Reply:
This test tests token that is attached to outgoing call from contract. VDelegatedCallRequest and VDelegatedCallResponse should be tested elsewhere with their own tests. From my point of view, testing all the messages in all the tests is a waste of time.